### PR TITLE
[Polymetis] Add impedance control macros to RobotInterface

### PR DESCRIPTION
**Pull request by exhaustin**
# Description

Add macros for updatable impedance control (joint & cartesian) to RobotInterface.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Added tests in `test/scripts`

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [x] I have made changes to existing documentation where needed.
- [x] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [x] [Polymetis only] I ran on hardware (1) that all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Comment by exhaustin** [comment]
Concerns:
- Do we need safety measures that prevents users from updating the desired state with something very far away from current states?
- Should we check whether the server is still running the impedance controller when updating? Pros: Handles cases where the impedance controller is terminated due to communication constraints, safety limits, other server issues, or another client overriding a command; Cons: Much updates due to an additional RPC call.

File: polymetis/polymetis/python/polymetis/robot_interface.py [hunk 1]
```diff
@@ -456,22 +461,104 @@ def move_to_ee_pose(
...
+
+        param_dict = {}
+        if position is not None:
+            param_dict["ee_pos_desired"] = position
+        if orientation is not None:
+            param_dict["ee_quat_desired"] = orientation
+
+        return self.update_current_policy(param_dict)
+
+    def terminate_impedance_controller(self):
```
Comment (1heart) [inline polymetis/polymetis/python/polymetis/robot_interface.py:554]: Do we need a separate of impedance controllers specifically?

**Review by 1heart — COMMENTED** [review]
IMO (1) is not within the scope of this PR, though it may be nice to have in the future.

(2) should be covered on the server side.

In general, I would prefer to have the check for whether a command be validated not through an `enum` on the client side (which could easily fall out of sync with the reality on the server), but rather from the server providing the source of truth (e.g. which type of controller is being run) and validating the client's attempted actions.

That doesn't mean that we need a separate RPC call to validate each `update` command -- I actually think (correct me if I'm wrong) an exception is already thrown if you attempt to update a field which doesn't exist. That exception can be surfaced on the client side when you attempt to update a controller incorrectly.

File: polymetis/polymetis/python/polymetis/robot_interface.py (hunk 1 above)
Comment (exhaustin) [reply polymetis/polymetis/python/polymetis/robot_interface.py:554]: ?

Comment (1heart) [reply polymetis/polymetis/python/polymetis/robot_interface.py:554]: Separate termination of impedance controllers*

Comment (exhaustin) [reply polymetis/polymetis/python/polymetis/robot_interface.py:554]: I don't think there needs to be. I think even having a single termination method for all non-blocking controllers would've been fine if we don't need to keep track of the mode on client side.

**Comment by exhaustin** [comment]
Hmm here's what happens currently if you try to update the server with a field that doesn't exist: The client side catches the exception but still freezes. The server doesn't print out any error messages but is unable to accept any other command, even after restarting the client.
I will try to fix the controller update error handling in a separate PR.

**Review by 1heart — APPROVED** [review]
Great, thank you!

**Pull request merged**